adding decky-recorder

marissa999 opened the pull request:
# Decky Recorder

This plugin allows the user to start and stop a recording of the screen + audio of the Steam Deck. This plugin is heavily based on the recapture-plugin for Crankshaft.

## Checklist:

### Developer Checklist

- [X] I am the original author or an authorized maintainer of this plugin.
- [X] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [X] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [X] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.

<!-- The following section needs to be modified as yes/no answers by the plugin developer. -->

<!-- Ex: "**Yes/No**: ..." becomes "**Yes**: ..." -->

### Plugin Backend Checklist

- **No**: I am using a custom backend other than Python.
Note: I do use the original gst-launcher-1.0 from SteamOS (which I call through Python), but I am downloading extra dependencies in a Holo-Base--Docker-Container and use those additional libraries as part of the gst-launcher-1.0-command
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **No**: I am using a custom binary that has all of it's dependencies statically linked.

<!-- The following section is for testers and maintainers, please do not modify it. -->

## Testing

- [ ] Tested on SteamOS Stable Update Channel.
- [ ] Tested on SteamOS Beta Update Channel.

Biyoni reviewed (approved):
Tested plugin on Stable, working well

TrainDoctor reviewed (commented):
Good to go to testing, submission approved.

marissa999 commented:
Oh, building failed because the entrypoint.sh is trying to clear up and delete a file that does not exist. Which normally wouldnt be an issue, but because ive used "set -e" it is failing. I will quickly fix that.

marissa999 commented:
Ok, should be working now.

I did a few changes on the main branch, including:
- Setting release version to 0.0.3
- Adding an option to record the mic audio
- Recordings get first saved to the /tmp (tmpfs) and then copied to the ~/Videos-Folder with FFmpeg (also repairs wrong timestamps in the video that gst seems to produce)
- Dependencies obtained through the dockerimage are now just blindly copied and not cherrypicked anymore, zip-file is now 7mb instead of 1.4mb but it will make it easier for me to add new features
- A lot of WIP-code that the user can not see right now

TrainDoctor commented:
@marissa999 you'll need to push the changes that allow the CI to work in order for me to rerun it.

marissa999 commented:
Oh yeah totally forgot about the submodule. I think I did it now?

TrainDoctor reviewed (changes requested):
Please make required changes to entrypoint script so that this submission can go to testing. Thanks.

EMERALD0874 commented:
Closing this PR since the creator has been unresponsive to requested changes that are preventing testing. Please reopen this PR once you have changes for testers to review.

Biyoni commented:
Hi,

I've run the tests on my own fork and this plugin builds correctly, can we rerun testing?

Thanks

lividhen commented:
@marissa999 New entry point commit works great on SteamOS Beta!

TrainDoctor commented:
@marissa999 Please let me know when you push a commit to this PR so that I can re-run CI. Submodules should now be properly recursively retrieved etc. Just make sure to rebase onto latest commits to main.

marissa999 commented:
I will, I still wanna try to either fix a bug first, that I introduced when I added Mic capture, or revert to an older version which has no mic. I also think that I need to adjust the backend outpath first. 

Also the build process seems to complain about pip3 missing while building the frontend. Which would be ok, I have no extra dependencies in the requirements.txt. Any idea what I could do to fix that?

TrainDoctor commented:
> Also the build process seems to complain about pip3 missing while building the frontend. Which would be ok, I have no extra dependencies in the requirements.txt. Any idea what I could do to fix that?

This was an error on my part, I accepted a PR to the plugin template that added a line that ran pip3 in the package.json.

Remove this line from package.json:
```
 "prepare": "touch requirements.txt && pip3 install --upgrade -r requirements.txt -t py_modules",
 ```

marissa999 commented:
> Closing this PR since the creator has been unresponsive to requested changes that are preventing testing. Please reopen this PR once you have changes for testers to review.

I think it should all work now, can you reopen this please? That would be amazing :D (maybe I can reopen it myself but i cant see it)

TrainDoctor commented:
> > Closing this PR since the creator has been unresponsive to requested changes that are preventing testing. Please reopen this PR once you have changes for testers to review.
> 
> I think it should all work now, can you reopen this please? That would be amazing :D (maybe I can reopen it myself but i cant see it)

Please push your changes to the PR. We are currently dealing with some CI related issues so we won't be able to properly deploy anything. Thanks for your patience.
